CesiumAsync/src/CachingAssetAccessor.cpp

```diff
@@ -311,17 +311,21 @@ bool shouldCacheRequest(
 
   // check if cache control contains no-store or no-cache directives
   int maxAge = 0;
+  int staleWhileRevalidate = 0;
   if (cacheControl) {
     if (cacheControl->noStore() || cacheControl->noCache()) {
       return false;
     }
 
     maxAge = cacheControl->maxAge();
+    staleWhileRevalidate = cacheControl->staleWhileRevalidate();
   }
 
-  // check response header contains expires if maxAge is not specified
-  const HttpHeaders& responseHeaders = pResponse->headers();
-  if (maxAge == 0) {
+  // Prefer cache control directives over older Expires header
+  if (maxAge != 0 || staleWhileRevalidate != 0) {
```

Comment:
I don't think we need to check for `staleWhileRevalidate`. Even with a `maxAge` of zero, it's still valid to _store_ the object as long as both of these are true:

1. It does _not_ have the `no-store` directive. It's fine if it has `no-cache`, because that just means you have to revalidate it each time.
2. It has an ETag or Date. In the former case we revalidate with `If-None-Match`, and in the latter we revalidate with `If-Modified-Since`.

If you haven't already seen it, this page is super helpful:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching

Basically, as long as we're not forbidden from storing it (`no-store`), and as long as we'll have _some_ way of revalidating it in the future, we should go ahead and store it in the cache because doing so is allowed and _may_ help us get a 304 from the server in the future.

Reply:
All good points. This first pass did just enough to get Google 3D tiles going to disk cache, but it looks prudent to go a bit further.

I've committed more changes, and good news, looks like any stale-while-revalidate logic isn't strictly needed, although I kept storage for it in `ResponseCacheControl`.

While the original problem of tiles not going to disk cache was caused by `max-age=0`, it was actually our implementation of `shouldCacheRequest` that was the problem. It would always return false for this response since the was no `Expires` header. This is point 2) in my PR description.

In addition, the root request had a slightly different issue which I noted in point 1). `max-age=0` and an old `Expires` value of `Mon, 01 Jan 1990 00:00:00 GMT`. This also caused `shouldCacheRequest` to always return false. Fixing this lets us cache our google sessionId across application runs, which I am very happy about.

Question for you... for rules that we follow in code that are backed by documentation, should we cite it in comments? For example, there is Mozilla documentation that states when to ignore the `Expires` header, which I am following.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expires

Reply:
> for rules that we follow in code that are backed by documentation, should we cite it in comments?